### PR TITLE
update fix ignored sort in api.py and add test

### LINKED ISSUE #43375: BUG: concat still sorts columns if they differ
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
>>> print(pd.concat([pd.DataFrame({i: i}, index=[i]) for i in range(10, 0, -1)], sort=False))
     2    3    4    5    6    7    8    9     10
10  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  10.0
9   NaN  NaN  NaN  NaN  NaN  NaN  NaN  9.0   NaN
8   NaN  NaN  NaN  NaN  NaN  NaN  8.0  NaN   NaN
7   NaN  NaN  NaN  NaN  NaN  7.0  NaN  NaN   NaN
6   NaN  NaN  NaN  NaN  6.0  NaN  NaN  NaN   NaN
5   NaN  NaN  NaN  5.0  NaN  NaN  NaN  NaN   NaN
4   NaN  NaN  4.0  NaN  NaN  NaN  NaN  NaN   NaN
3   NaN  3.0  NaN  NaN  NaN  NaN  NaN  NaN   NaN
2   2.0  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN
```

#### Problem description

This was marked as fixed in #4588, but `concat()` still sorts the columns if they differ. The `sort` argument doesn't seem to have any effect on this.

#### Expected Output
```python
     10   9    8    7    6    5    4    3     2 
10  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  10.0
9   NaN  NaN  NaN  NaN  NaN  NaN  NaN  9.0   NaN
8   NaN  NaN  NaN  NaN  NaN  NaN  8.0  NaN   NaN
7   NaN  NaN  NaN  NaN  NaN  7.0  NaN  NaN   NaN
6   NaN  NaN  NaN  NaN  6.0  NaN  NaN  NaN   NaN
5   NaN  NaN  NaN  5.0  NaN  NaN  NaN  NaN   NaN
4   NaN  NaN  4.0  NaN  NaN  NaN  NaN  NaN   NaN
3   NaN  3.0  NaN  NaN  NaN  NaN  NaN  NaN   NaN
2   2.0  NaN  NaN  NaN  NaN  NaN  NaN  NaN   NaN
```
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 5f648bf1706dd75a9ca0d29f26eadfbb595fe52b
python           : 3.8.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-77-generic
Version          : #86-Ubuntu SMP Thu Jun 17 02:35:03 UTC 2021
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 1.3.2
numpy            : 1.21.2
pytz             : 2021.1
dateutil         : 2.8.2
pip              : 20.0.2
setuptools       : 44.0.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by CloseChoice**
- [x] closes #43375 
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

As mentioned in the ticket, the function `api.union_indexes` does not handle the sort parameter. I added support for it, but we explicitly check that sort can only take the values `False` and `None` in [here](https://github.com/pandas-dev/pandas/blob/62d1f5c0fb16b6990aceabb8491d61de63cd042a/pandas/core/indexes/base.py#L2898)

### CODE REVIEW

File: pandas/tests/arrays/integer/test_concat.py [hunk 1]
```diff
@@ -63,3 +63,13 @@ def test_concat_series_with_numpy(to_concat_dtypes, result_dtype):
     result = pd.concat([s2, s1], ignore_index=True)
     expected = pd.Series([0, 1, 0, 1, pd.NA], dtype=object).astype(result_dtype)
     tm.assert_series_equal(result, expected)
+
+
+def test_concat_frame_with_sort_false():
```
Comment (jreback) [inline pandas/tests/arrays/integer/test_concat.py:68]: can you move to pandas/tests/reshape/concat/test_sort.py

put near simliar tests

**Review by jreback — CHANGES_REQUESTED** [review]
might have to adjust some other tests

**Comment by CloseChoice** [comment]
> might have to adjust some other tests

Can we just adjust other test results? That would be technically be a breaking change, wouldn't it?

**Comment by jreback** [comment]
my point about adjusting test is that this case might be only internal or not - will have to see

**Review by jreback — CHANGES_REQUESTED** [review]
looks good, can you add a whatsnew note for 1.4, reshaping and ping on green

**Comment by CloseChoice** [comment]
@github-actions pre-commit

**Comment by CloseChoice** [comment]
@jreback all lights on green.

**Review by jreback — APPROVED** [review]

**Pull request merged**